sdk/identity/Azure.Identity/src/OnBehalfOfCredentialOptions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Identity
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class OnBehalfOfCredentialOptions : TokenCredentialOptions, ITokenCacheOptions
+    {
+        /// <summary>
+        /// The <see cref="TokenCachePersistenceOptions"/>.
+        /// </summary>
+        public TokenCachePersistenceOptions TokenCachePersistenceOptions { get; set; }
+
+        /// <summary>
+        /// A client secret that was generated for the App Registration used to authenticate the client.
+        /// </summary>
+        public string CertificatePath { get; set; }
+
+        /// <summary>
+        /// Will include x5c header in client claims when acquiring a token to enable subject name / issuer based authentication for the <see cref="ClientCertificateCredential"/>.
+        /// </summary>
+        public bool SendCertificateChain { get; set; }
```

Comment:
Should we throw if this option is set and the `OnBehalfOfCredential`  is being constructed with a client secret? If we don't will MSAL throw when we request the token?

Reply:
we go down a different constructor path that doesn't read that option if you use the client secret public ctor.